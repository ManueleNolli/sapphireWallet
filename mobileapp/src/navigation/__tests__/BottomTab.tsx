import React from 'react'
import { NavigationContainer } from '@react-navigation/native'
import BottomTabNavigator from '../BottomTab'
import renderWithTheme from '../../TestHelper'
import { act, fireEvent, waitFor } from '@testing-library/react-native'

// MOCKS
jest.mock('../../pages/Home/Home')
jest.mock('../../pages/Details/Details')
jest.mock('../../pages/Settings/Settings')

describe('BottomTabNavigator', () => {
  it('Renders correctly first page', () => {
    const tree = renderWithTheme(
      <NavigationContainer>
        <BottomTabNavigator />
      </NavigationContainer>
    )

    // Assert that the initial screen is rendered
    expect(tree.getAllByText('Home')).toBeDefined()
    expect(tree.getByText('NFTs')).toBeDefined()
    expect(tree.getByText('Settings')).toBeDefined()
    expect(tree.toJSON()).toMatchSnapshot()
  })

  it('Navigates to Settings when pressing the "Settings" tab', async () => {
    const tree = renderWithTheme(
      <NavigationContainer>
        <BottomTabNavigator />
      </NavigationContainer>
    )

    await act(async () => {
      const button = await tree.getByText('Settings')
      fireEvent.press(button)
      await waitFor(() => {
        expect(tree.getAllByText('Settings')).toBeDefined()
      })
    })
  })

  it('Navigates to Settings when pressing the "NFTs" tab', async () => {
    const tree = renderWithTheme(
      <NavigationContainer>
        <BottomTabNavigator />
      </NavigationContainer>
    )

    await act(async () => {
      const button = await tree.getByText('NFTs')
      fireEvent.press(button)
      await waitFor(() => {
        expect(tree.getAllByText('NFTs')).toBeDefined()
      })
    })
  })
})
