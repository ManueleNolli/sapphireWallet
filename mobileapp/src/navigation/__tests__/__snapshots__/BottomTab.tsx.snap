// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BottomTabNavigator Renders correctly first page 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenNavigationContainer
    style={
      {
        "flex": 1,
        "overflow": "hidden",
      }
    }
  >
    <RNSScreen
      activityState={2}
      collapsable={false}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      sheetAllowedDetents="large"
      sheetCornerRadius={-1}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetLargestUndimmedDetent="all"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": 0,
        }
      }
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          [
            {
              "backgroundColor": "rgb(242, 242, 242)",
              "flex": 1,
            },
            [
              {
                "flex": 1,
                "flexDirection": "column-reverse",
              },
              undefined,
            ],
          ]
        }
      >
        <View
          style={
            {
              "flex": 1,
            }
          }
        >
          <div
            data-testid="Home"
          >
            Home
          </div>
        </View>
        <View
          onLayout={[Function]}
          style={
            {
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            pointerEvents="box-none"
            style={
              {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "zIndex": 0,
              }
            }
          />
          <View
            collapsable={false}
            pointerEvents="box-none"
            style={
              {
                "height": 44,
                "maxHeight": undefined,
                "minHeight": undefined,
                "opacity": undefined,
                "transform": undefined,
              }
            }
          >
            <View
              pointerEvents="none"
              style={
                {
                  "height": 0,
                }
              }
            />
            <View
              pointerEvents="box-none"
              style={
                {
                  "alignItems": "stretch",
                  "flex": 1,
                  "flexDirection": "row",
                }
              }
            >
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  {
                    "alignItems": "flex-start",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "justifyContent": "center",
                    "marginStart": 0,
                  }
                }
              >
                <ViewManagerAdapter_ExpoImage
                  contentFit="contain"
                  contentPosition={
                    {
                      "left": "50%",
                      "top": "50%",
                    }
                  }
                  height={32}
                  marginLeft={8}
                  onError={[Function]}
                  onLoad={[Function]}
                  onLoadStart={[Function]}
                  onProgress={[Function]}
                  placeholder={[]}
                  source={[]}
                  style={
                    {
                      "height": 32,
                      "marginLeft": 8,
                      "width": 32,
                    }
                  }
                  transition={null}
                  width={32}
                />
              </View>
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  {
                    "justifyContent": "center",
                    "marginHorizontal": 16,
                    "maxWidth": 590,
                  }
                }
              >
                <Text
                  appearance="default"
                  category="h6"
                  status="basic"
                  style={
                    [
                      {
                        "color": "#222B45",
                        "fontFamily": "System",
                        "fontSize": 18,
                        "fontWeight": "800",
                      },
                      undefined,
                    ]
                  }
                >
                   
                  Home
                   
                </Text>
              </View>
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  {
                    "alignItems": "flex-end",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "justifyContent": "center",
                    "marginEnd": 0,
                  }
                }
              >
                <ViewManagerAdapter_ExpoImage
                  contentFit="contain"
                  contentPosition={
                    {
                      "left": "50%",
                      "top": "50%",
                    }
                  }
                  height={32}
                  marginRight={8}
                  onError={[Function]}
                  onLoad={[Function]}
                  onLoadStart={[Function]}
                  onProgress={[Function]}
                  placeholder={[]}
                  source={[]}
                  style={
                    {
                      "height": 32,
                      "marginRight": 8,
                      "width": 32,
                    }
                  }
                  transition={null}
                  width={32}
                />
              </View>
            </View>
          </View>
        </View>
      </View>
    </RNSScreen>
  </RNSScreenNavigationContainer>
  <View>
    <View
      collapsable={false}
      onLayout={[Function]}
      positions={3}
      selectedPosition={0}
      style={
        {
          "transform": [
            {
              "translateX": 0,
            },
          ],
          "width": "33.333333333333336%",
        }
      }
    >
      <View
        style={
          [
            {
              "alignSelf": "center",
              "width": "100%",
            },
            {
              "backgroundColor": "#3366FF",
              "height": 4,
            },
          ]
        }
        testID="indicator body"
      />
    </View>
    <View
      style={
        [
          {
            "backgroundColor": "#FFFFFF",
            "minHeight": 56,
            "paddingVertical": 4,
          },
          {
            "flexDirection": "row",
          },
          {
            "paddingBottom": 26.68,
          },
        ]
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
            "opacity": 1,
          }
        }
      >
        <View
          collapsable={false}
          style={
            {
              "transform": [
                {
                  "scale": 1,
                },
              ],
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight="24"
            bbWidth="24"
            fill="#3366FF"
            focusable={false}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                {
                  "height": 24,
                  "marginVertical": 2,
                  "tintColor": "#3366FF",
                  "width": 24,
                },
                {
                  "flex": 0,
                  "height": 24,
                  "width": 24,
                },
              ]
            }
            vbHeight={24}
            vbWidth={24}
          >
            <RNSVGGroup
              fill={
                {
                  "payload": 4281558783,
                  "type": 0,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            >
              <RNSVGGroup
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
              >
                <RNSVGGroup
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                >
                  <RNSVGRect
                    fill={
                      {
                        "payload": 4278190080,
                        "type": 0,
                      }
                    }
                    height="24"
                    opacity={0}
                    width="24"
                    x="0"
                    y="0"
                  />
                  <RNSVGPath
                    d="M12 11a4 4 0 1 0-4-4 4 4 0 0 0 4 4z"
                    fill={
                      {
                        "payload": 4278190080,
                        "type": 0,
                      }
                    }
                  />
                  <RNSVGPath
                    d="M18 21a1 1 0 0 0 1-1 7 7 0 0 0-14 0 1 1 0 0 0 1 1z"
                    fill={
                      {
                        "payload": 4278190080,
                        "type": 0,
                      }
                    }
                  />
                </RNSVGGroup>
              </RNSVGGroup>
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
        <Text
          appearance="default"
          category="p1"
          status="basic"
          style={
            [
              {
                "color": "#222B45",
                "fontFamily": "System",
                "fontSize": 15,
                "fontWeight": "400",
              },
              {
                "color": "#3366FF",
                "fontFamily": "System",
                "fontSize": 12,
                "fontWeight": "600",
                "marginVertical": 2,
              },
            ]
          }
        >
          Home
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
            "opacity": 1,
          }
        }
      >
        <View
          collapsable={false}
          style={
            {
              "transform": [
                {
                  "scale": 1,
                },
              ],
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight="24"
            bbWidth="24"
            fill="#8F9BB3"
            focusable={false}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                {
                  "height": 24,
                  "marginVertical": 2,
                  "tintColor": "#8F9BB3",
                  "width": 24,
                },
                {
                  "flex": 0,
                  "height": 24,
                  "width": 24,
                },
              ]
            }
            vbHeight={24}
            vbWidth={24}
          >
            <RNSVGGroup
              fill={
                {
                  "payload": 4287601587,
                  "type": 0,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            >
              <RNSVGGroup
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
              >
                <RNSVGGroup
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                >
                  <RNSVGRect
                    fill={
                      {
                        "payload": 4278190080,
                        "type": 0,
                      }
                    }
                    height="24"
                    opacity={0}
                    width="24"
                    x="0"
                    y="0"
                  />
                  <RNSVGPath
                    d="M18 3H6a3 3 0 0 0-3 3v12a3 3 0 0 0 3 3h12a3 3 0 0 0 3-3V6a3 3 0 0 0-3-3zM6 5h12a1 1 0 0 1 1 1v8.36l-3.2-2.73a2.77 2.77 0 0 0-3.52 0L5 17.7V6a1 1 0 0 1 1-1z"
                    fill={
                      {
                        "payload": 4278190080,
                        "type": 0,
                      }
                    }
                  />
                  <RNSVGCircle
                    cx="8"
                    cy="8.5"
                    fill={
                      {
                        "payload": 4278190080,
                        "type": 0,
                      }
                    }
                    r="1.5"
                  />
                </RNSVGGroup>
              </RNSVGGroup>
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
        <Text
          appearance="default"
          category="p1"
          status="basic"
          style={
            [
              {
                "color": "#222B45",
                "fontFamily": "System",
                "fontSize": 15,
                "fontWeight": "400",
              },
              {
                "color": "#8F9BB3",
                "fontFamily": "System",
                "fontSize": 12,
                "fontWeight": "600",
                "marginVertical": 2,
              },
            ]
          }
        >
          NFTs
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
            "opacity": 1,
          }
        }
      >
        <View
          collapsable={false}
          style={
            {
              "transform": [
                {
                  "scale": 1,
                },
              ],
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight="24"
            bbWidth="24"
            fill="#8F9BB3"
            focusable={false}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                {
                  "height": 24,
                  "marginVertical": 2,
                  "tintColor": "#8F9BB3",
                  "width": 24,
                },
                {
                  "flex": 0,
                  "height": 24,
                  "width": 24,
                },
              ]
            }
            vbHeight={24}
            vbWidth={24}
          >
            <RNSVGGroup
              fill={
                {
                  "payload": 4287601587,
                  "type": 0,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            >
              <RNSVGGroup
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
              >
                <RNSVGGroup
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                >
                  <RNSVGRect
                    fill={
                      {
                        "payload": 4278190080,
                        "type": 0,
                      }
                    }
                    height="24"
                    opacity={0}
                    width="24"
                    x="0"
                    y="0"
                  />
                  <RNSVGCircle
                    cx="12"
                    cy="12"
                    fill={
                      {
                        "payload": 4278190080,
                        "type": 0,
                      }
                    }
                    r="1.5"
                  />
                  <RNSVGPath
                    d="M21.89 10.32L21.1 7.8a2.26 2.26 0 0 0-2.88-1.51l-.34.11a1.74 1.74 0 0 1-1.59-.26l-.11-.08a1.76 1.76 0 0 1-.69-1.43v-.28a2.37 2.37 0 0 0-.68-1.68 2.26 2.26 0 0 0-1.6-.67h-2.55a2.32 2.32 0 0 0-2.29 2.33v.24a1.94 1.94 0 0 1-.73 1.51l-.13.1a1.93 1.93 0 0 1-1.78.29 2.14 2.14 0 0 0-1.68.12 2.18 2.18 0 0 0-1.12 1.33l-.82 2.6a2.34 2.34 0 0 0 1.48 2.94h.16a1.83 1.83 0 0 1 1.12 1.22l.06.16a2.06 2.06 0 0 1-.23 1.86 2.37 2.37 0 0 0 .49 3.3l2.07 1.57a2.25 2.25 0 0 0 1.35.43A2 2 0 0 0 9 22a2.25 2.25 0 0 0 1.47-1l.23-.33a1.8 1.8 0 0 1 1.43-.77 1.75 1.75 0 0 1 1.5.78l.12.17a2.24 2.24 0 0 0 3.22.53L19 19.86a2.38 2.38 0 0 0 .5-3.23l-.26-.38A2 2 0 0 1 19 14.6a1.89 1.89 0 0 1 1.21-1.28l.2-.07a2.36 2.36 0 0 0 1.48-2.93zM12 15.5a3.5 3.5 0 1 1 3.5-3.5 3.5 3.5 0 0 1-3.5 3.5z"
                    fill={
                      {
                        "payload": 4278190080,
                        "type": 0,
                      }
                    }
                  />
                </RNSVGGroup>
              </RNSVGGroup>
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
        <Text
          appearance="default"
          category="p1"
          status="basic"
          style={
            [
              {
                "color": "#222B45",
                "fontFamily": "System",
                "fontSize": 15,
                "fontWeight": "400",
              },
              {
                "color": "#8F9BB3",
                "fontFamily": "System",
                "fontSize": 12,
                "fontWeight": "600",
                "marginVertical": 2,
              },
            ]
          }
        >
          Settings
        </Text>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
